codegen/generators/types-gen/generator/types.go

```diff
@@ -157,3 +171,14 @@ func GenerateMirTypes(inputDir, sourcePackagePath string, msgs []*types.Message,
 
 	return codegen.RenderJenFile(jenFile, types.OutputDir(inputDir), "types.mir.go")
 }
+
+// canBeNil returns true if the field can be nil.
+// at the moment, Maps and Slices cannot be nil by this check, as we use an internal conversion to our types.Slice and types.Map
```

Comment:
They can be nil, but I mean in this comment that currently the code checks that elsewhere (particularly, in the ConvertMap and ConvertSlice functions in their respective files at `codegen/model/types/`). This is because the codegen internal treatment of Maps and Slices means that it converts their reflection `.Kind()` to a struct, so this check will never detect maps or slices, but I think it is good practice to be comprehensive here in case we change further the codegen later.